Intel Fortran release mode uses different calling convention

fpm sets `-assume nounderscore` if the `--release` flag is provided, but does not set this option for the default debug build. This means LAPACK / BLAS routines can only be called with Intel in debug mode right now.

```
 + ifort  -fp-model precise -pc 64 -align all -error-limit 1 -reentrancy threaded -nogen-interfaces -assume byterecl -assume nounderscore  -module build/ifort_release/multicharge -I build/ifort_release/multicharge build/ifort_release/multicharge/app_main.f90.o build/ifort_release/multicharge/libmulticharge.a  -llapack -lblas -llapack -lblas -o build/ifort_release/app/multicharge
ld: build/ifort_release/multicharge/libmulticharge.a(src_multicharge_blas.f90.o): in function `multicharge_blas_mp_mchrg_sgemv312':
blas.f90:(.text+0x38a): undefined reference to `sgemv'
...
```

--- a/fpm/src/fpm.f90
+++ b/fpm/src/fpm.f90
@@ -69,6 +69,9 @@ subroutine build_model(model, settings, package, error)
     model%output_directory = join_path('build',basename(model%fortran_compiler)//'_'//settings%build_name)
 
     call add_compile_flag_defaults(settings%build_name, basename(model%fortran_compiler), model)
+    if(settings%verbose)then
+       write(*,*)'<INFO>COMPILER OPTIONS:  ', model%fortran_compile_flags 
+    endif
 
     model%link_flags = ''
 

--- a/fpm/src/fpm_compiler.f90
+++ b/fpm/src/fpm_compiler.f90
@@ -12,7 +12,7 @@ type(fpm_model_t), intent(inout) :: model
 ! could just be a function to return a string instead of passing model
 ! but likely to change other components like matching C compiler
 
-character(len=:),allocatable :: fflags  ! optional flags that might be overridden by user
+character(len=:),allocatable :: fflags    ! optional flags that might be overridden by user
 character(len=:),allocatable :: modpath 
 character(len=:),allocatable :: mandatory ! flags required for fpm to function properly;
                                           ! ie. add module path and module include directory as appropriate
@@ -42,6 +42,24 @@ character(len=:),allocatable :: mandatory ! flags required for fpm to function p
 ! G95               ?          ?       -fmod=          -I            -fopenmp   discontinued
 ! Open64            ?          ?       -module         -I            -mp        discontinued
 ! Unisys            ?          ?       ?               ?             ?          discontinued
+character(len=*),parameter :: names(*)=[ character(len=10) :: &
+& 'caf', &
+& 'gfortran', &
+& 'f95', &
+& 'nvfortran', &
+& 'ifort', &
+& 'ifx', &
+& 'pgfortran', &
+& 'pgf90', &
+& 'pgf95', &
+& 'flang', &
+& 'lfc', &
+& 'nagfor', &
+& 'crayftn', &
+& 'xlf90', &
+& 'unknown']
+integer :: i
+
     modpath=join_path(model%output_directory,model%package_name)
     fflags=''
     mandatory=''
@@ -143,7 +161,6 @@ character(len=:),allocatable :: mandatory ! flags required for fpm to function p
        & -reentrancy threaded&
        & -nogen-interfaces&
        & -assume byterecl&
-       & -assume nounderscore&
        &'
        mandatory=' -module '//modpath//' -I '//modpath 
     case('debug_ifort')
@@ -219,10 +236,8 @@ character(len=:),allocatable :: mandatory ! flags required for fpm to function p
     case default
        fflags = ' '
        mandatory=' -module '//modpath//' -I '//modpath 
-       write(*,*)'<WARNING> unknown compiler (',compiler,')'
-       write(*,*)'          and build name (',build_name,')'
-       write(*,*)'          combination.'
-       write(*,*)'          known compilers are gfortran, nvfortran, ifort'
+       write(*,'(*(a))')'<WARNING> unknown compiler (',compiler,') and build name (',build_name,') combination.'
+       write(*,'(a,*(T31,6(a:,", "),/))')'          known compilers are ',(trim(names(i)),i=1,size(names)-1)
     end select
 
     model%fortran_compile_flags = fflags//' '//mandatory

